lib/peer.go

```diff
@@ -180,7 +180,7 @@ func (pp *Peer) HandleGetTransactionsMsg(getTxnMsg *MsgDeSoGetTransactions) {
 		// If the transaction isn't in the pool, just continue without adding
 		// it. It is generally OK to respond with only a subset of the transactions
 		// that were requested.
-		if mempoolTx == nil {
+		if mempoolTx == nil || !mempoolTx.IsValidated() {
```

Comment:
Nice. I think this should prevent us from sending unvalidated txns on the first sync which is good.